demo/docker/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.6
+
+ENV USER=root
+
+# Upgrade or install packages needed
+RUN apt-get update && apt-get -y install \
+    g++ \
+    gcc \
+    binutils \
+    clang \ 
+    cmake 
+
+# Install xlearn
+RUN git clone https://github.com/aksnzhy/xlearn.git 
+RUN cd xlearn && ./build.sh
+# Install liblinear
+RUN cd ..
+RUN git clone https://github.com/cjlin1/liblinear.git
+RUN cd liblinear/python/ && make
+
+
+ENTRYPOINT ["/bin/bash"]
```

Comment:
we can also reduce the size of the image by making use of [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/), as it would allow us to not have to include the compiler and all the auxiliary files that get created when the libraries get installed.

For this we need to add a name to the first image:

```dockerfile
FROM python:3.7-slim-buster AS base

RUN ...
# all the commands you have above that install the libraries, then:


FROM python:3.7-slim-buster AS lib
COPY --from base /usr/local/lib/python3.7/site-packages /usr/local/lib/python3.7/
```

the last two lines say something like: "create a new layer from the `python:3.7-slim-buster` image, and in that image copy from the `base` image the site-packages folder".